internal/corazawaf/rule.go

```diff
@@ -155,11 +158,11 @@ func (r *Rule) Status() int {
 // Evaluate will evaluate the current rule for the indicated transaction
 // If the operator matches, actions will be evaluated, and it will return
 // the matched variables, keys and values (MatchData)
-func (r *Rule) Evaluate(tx rules.TransactionState) []types.MatchData {
-	return r.doEvaluate(tx.(*Transaction))
+func (r *Rule) Evaluate(tx rules.TransactionState, cache map[transformationKey]transformationValue) []types.MatchData {
```

Comment:
This change makes me wonder if evaluate should be exported or not as it is only used inside this package. Does not need to happen in this PR but this is a good justification cc @jptosso @anuraaga @M4tteoP